apps/crn-server/test/integration/fixtures/types.ts

```diff
@@ -0,0 +1,59 @@
+import {
+  UserCreateDataObject as CommonUserCreateDataObject,
+  TeamCreateDataObject as CommonTeamCreateDataObject,
+  ResearchOutputCreateDataObject as CommonResearchOutputCreateDataObject,
+  GroupDataObject,
+  GroupRole,
+} from '@asap-hub/model';
+
+export type UserCreateDataObject = Omit<CommonUserCreateDataObject, 'labIds'>;
+export type TeamCreateDataObject = CommonTeamCreateDataObject;
+export type ResearchOutputCreateDataObject = Omit<
+  CommonResearchOutputCreateDataObject,
+  | 'teamIds'
+  | 'methodIds'
+  | 'organismIds'
+  | 'environmentIds'
+  | 'keywordIds'
+  | 'labIds'
+  | 'createdBy'
+> & {
+  teams: string[];
+  published: boolean;
+  methods: string[];
+  organisms: string[];
+  environments: string[];
+  keywords: string[];
+  labs: string[];
+  authors: { userId: string }[];
+};
+export type InterestGroupCreateDataObject = Omit<
+  GroupDataObject,
+  | 'id'
+  | 'lastModifiedDate'
+  | 'createdDate'
+  | 'leaders'
+  | 'teams'
+  | 'contactEmails'
+  | 'calendars'
+  | 'tools'
+> & {
+  leaders: { user: string; role: GroupRole; inactiveSinceDate?: string }[];
+  teams: { id: string }[];
+  calendar: { id: string } | null;
+};
+
+type EntryFixture<T> = T & { id: string };
+
+export type UserFixture = EntryFixture<UserCreateDataObject>;
+export type TeamFixture = EntryFixture<TeamCreateDataObject>;
+export type InterestGroupFixture = EntryFixture<InterestGroupCreateDataObject>;
+
+export interface Fixture {
+  createUser: (user: UserCreateDataObject) => Promise<UserFixture>;
```

Comment:
its a little like we were avoiding to create data-provider methods which would then have to maintain later but then doing that anyway, not only is the input interface very similar (and extended from) the original data provider methods - it reuses some of the same functions that the data provider does which means the implementation is probably similar, but that's fine, im only showing you why i think we could also just extend data providers

Reply:
yeah you see this is my only point. In my mind thats a lot more overhead - having to maintain two implementations. We do introduce a new set of types but since you are referencing the original ones we might actually be able to avoid accidentally forgetting to update these fixtures. 

This is my example - say we introduce a new, required field - I want to make sure that when we update the types in a single place this automatically gets flagged anywhere where the new addition is necessary (for instance in fixtures). I think with how you have done it will get picked up so we're good